src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,138 @@
+        String keyword = line.split(" ")[0].toLowerCase();
+
+        if (!keyword.equals("list") && !keyword.equals("bye")
+        && !keyword.equals("todo") && !keyword.equals("done")
+        && !keyword.equals("event") && !keyword.equals("deadline")){
+            throw new DukeCheckLineException();
+        }
+    }
+
+    static void checkDescription(String line)throws DukeException{
+        String keyword = line.split(" ")[0].toLowerCase();
+
+        if (keyword.equals("todo") && line.split(" ").length == 1){
+            throw new DukeException();
+        }
+
+        if (keyword.equals("event") && line.split(" ").length == 1){
+            throw new DukeException();
+        }
+
+        if (keyword.equals("deadline") && line.split(" ").length == 1){
+            throw new DukeException();
+        }
+
+        if (keyword.equals("done") && line.split(" ").length == 1){
+            throw new DukeException();
+        }
+    }
+
+    public static void main(String[] args){
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+
+        printWithLine(List.of("Hello from\n" + logo));
+        printWithLine(List.of("Hello! I'm Duke", "What can I do for you?"));
+
+        String line;
+        Scanner in = new Scanner(System.in);
+        List<Task> items = new ArrayList<>();
+
+        do{
+            line = in.nextLine();
+            String keyword = line.split(" ")[0].toLowerCase();
+
+            try {
+                checkLineEmpty(line);
+                checkWord(line);
+                checkDescription(line);
+
+                if (line.equalsIgnoreCase("list")){
+                    printWithLine(List.of());
+                    List<String> messages = new ArrayList<>();
+                    System.out.println("   Here are the tasks in your list: ");
+                    for (int i = 0; i < items.size(); i++) {
+                        messages.add(i + 1 + "." + items.get(i));
+                    }
+                    printWithLine(messages);
+
+                } else if (line.split(" ")[0].equalsIgnoreCase("done")){
+                    printWithLine(List.of());
+                    Task markItem = items.get(Integer.parseInt(line.substring(5)) - 1);
+                    markItem.markAsDone();
+                    printWithLine(List.of("Nice! I've marked this task as done: ", " " + markItem));
+
+                } else if (line.split(" ")[0].equalsIgnoreCase("todo")){
+                    printWithLine(List.of());
+                    Task todoTask = new Todo(line.replace(line.split(" ")[0] + " ", ""));
+                    items.add(todoTask);
+                    printWithLine((List.of("Got it. I've added this task: ", todoTask.toString(), "Now you have " + items.size() + " task in the list. ")));
```

Comment:
Put inside a method to handle this else if block for readers to understand the flow clearly.

Reply:
Thanks for your advice, will update the code.